Test container example for ImmuDB

hariohmprasath opened the pull request:
## Description of the change:

Added an example for running tests using `ImmuDB` database using `testcontainers`. Tests performs all the basic operations performed on the `ImmuDB` #5594 

We already have a ask for this example, so contributed the same using this pull request:
https://github.com/testcontainers/testcontainers-java/issues/5594

<!--
Thanks for contributing to Testcontainers. Please review the following notes before
submitting a pull request.

New Modules:
Make sure to add it to `bug_report.yaml`, `enhancement.yaml` and `feature.yaml`.
Also, add it to `dependabot.yml` and `labeler.yml`.

Before comitting, run `./gradlew checkstyleMain checkstyleTest spotlessApply` and fix any issues that occur.

Describing Your Changes:

If, after having reviewed the notes above, you're ready to submit your pull request, please
provide a brief description of the proposed changes and their context. If they fix a bug, please
describe the broken behaviour and how the changes fix it. If they make an enhancement,
please describe the new functionality and why you believe it's useful. If your pull
request relates to any existing issues, please reference them by using the issue number
prefixed with #.
-->

eddumelendez commented on examples/immudb/src/test/java/ImmuDbTest.java:
```diff
@@ -0,0 +1,171 @@
+    /**
+     * Test method for the getValue.
+     */
+    @Test
+    public void testGetValue() {
+        try {
+            this.immuDbClient.putValue("test1", "test2");
+            Assert.assertEquals("test2", this.immuDbClient.getValue("test1"));
+        }
+        catch (CorruptedDataException | VerificationException e) {
+            Assert.fail();
+        }
+    }
```
we can keep this test. Given the purpose is to demonstrate how to start the container and build the client, this test will be more than enough. Also, use `assertThat` from `Assertions` (assertj)

eddumelendez commented on examples/immudb/build.gradle:
```diff
@@ -0,0 +1,15 @@
+    compileOnly 'org.slf4j:slf4j-api:2.0.0'
+    implementation 'io.codenotary:immudb4j:0.9.10.2'
+    testImplementation 'org.testcontainers:testcontainers'
+    testImplementation 'org.testng:testng:7.5'
```
remove this dependency

eddumelendez reviewed (changes requested):
thanks for the PR @hariohmprasath ! Few notes, `ImmuDbClient` and `ImmuDbClientImpl` can be removed and the client can be used directly in the test class. Also, add the example [here](https://github.com/testcontainers/testcontainers-java/blob/master/docs/examples.md)

hariohmprasath commented:
> thanks for the PR @hariohmprasath ! Few notes, `ImmuDbClient` and `ImmuDbClientImpl` can be removed and the client can be used directly in the test class. Also, add the example [here](https://github.com/testcontainers/testcontainers-java/blob/master/docs/examples.md)

Thanks @eddumelendez for the review. I have addressed all the feedback, let me know if you any suggestions. Thanks

eddumelendez commented:
Add the example [here](https://github.com/testcontainers/testcontainers-java/blob/master/docs/examples.md), please.

hariohmprasath commented:
> Add the example [here](https://github.com/testcontainers/testcontainers-java/blob/master/docs/examples.md), please.

Sorry about missing that in my previous commit. I have updated now, so we should good. Thanks

eddumelendez commented on examples/immudb/src/test/java/ImmuDbTest.java:
```diff
@@ -0,0 +1,66 @@
+
+    // Test container for the ImmuDb database, with the latest version of the image and exposed port
+    @ClassRule
+    public static final GenericContainer<?> immuDbContainer = new GenericContainer<>("codenotary/immudb:latest")
```
last one... do not use latest :)

hariohmprasath replied:
No problem, updated the image tag to 1.3

eddumelendez commented on examples/immudb/src/test/java/ImmuDbTest.java:
```diff
@@ -0,0 +1,66 @@
+    // ImmuClient used to interact with the DB
+    private ImmuClient immuClient;
+
+    /**
+     * Set up the test, by creating the ImmuDbClient and connecting to the DB.
+     */
```
```suggestion
```

eddumelendez commented:
Thanks for your contribution, @hariohmprasath ! This is not in `main` branch
